src/Admin/Controllers/HomeController.cs

```diff
@@ -42,26 +44,19 @@ public IActionResult Error()
             });
         }
 
-        public async Task<IActionResult> GetLatestDockerHubVersion(string repository, CancellationToken cancellationToken)
+        public async Task<IActionResult> GetLatestVersion(string project, CancellationToken cancellationToken)
         {
-            var requestUri = $"https://hub.docker.com/v2/repositories/bitwarden/{repository}/tags/";
+            var requestUri = $"https://raw.githubusercontent.com/bitwarden/self-host/master/version.json";
             try
             {
                 var response = await _httpClient.GetAsync(requestUri, cancellationToken);
                 if (response.IsSuccessStatusCode)
                 {
                     using var jsonDocument = await JsonDocument.ParseAsync(await response.Content.ReadAsStreamAsync(cancellationToken), cancellationToken: cancellationToken);
                     var root = jsonDocument.RootElement;
-
-                    var results = root.GetProperty("results");
-                    foreach (var result in results.EnumerateArray())
-                    {
-                        var name = result.GetProperty("name").GetString();
-                        if (!string.IsNullOrWhiteSpace(name) && name.Length > 0 && char.IsNumber(name[0]))
-                        {
-                            return new JsonResult(name);
-                        }
-                    }
+                    var versionsNode = root.GetProperty("versions");
+                    var version = versionsNode.GetProperty($"{project}Version").GetString();
```

Comment:
Will this throw an error that we need to catch?

We could also get rid of json document and serialize it to a class since this data model is more structured than the docker was.

Reply:
true, changing it.